### PR TITLE
Verify date of merges when pulling list of changes

### LINKED ISSUE #141: Automation gets into a loop
Current code in util/automation.py requests changes with content from last_sha1 file against the current refs/heads/wer-tracking.

Starting with https://github.com/mozilla/DeepSpeech/compare/340d31ae9c80fbeedc5a810452267f139b014d1f...refs/heads/wer-tracking this shows some commits "in the past", hence making the automation somehow looping :)

### PR DESCRIPTION
**Pull request by lissyx**
Fixes #141

### CODE REVIEW

**Comment by lissyx** [comment]
This is now the output that I get:
> Existing SHA1: 340d31ae9c80fbeedc5a810452267f139b014d1f fetching changes
compare_url= https://api.github.com/repos/mozilla/DeepSpeech/compare/340d31ae9c80fbeedc5a810452267f139b014d1f...refs/heads/wer-tracking
> Will execute for [u'cba81fb5e42ca1963a45e177a0cfe64ffed4e3c4', u'e9c9103a1ef11734273cf59317f6f7986615fa63', u'4add35b81bdcfe4c5215d78d71b6dbaabec70038', u'529766a0807c2599c589b99f7e203ae2899a9eb1', u'5f58d57e62e9493ac78225ebddc1e86bb5b813d3', u'5fad4e30037c298e92dcb5a1cbc31f6a79070fdc', u'f4766cce92131b01604609de338103cc24d67c76', u'6449fc45bdfd2da369fd9757a0511452cb3859eb', u'b56e26c51d9a2c1dca5cacac924400ac7d05a27b']

File: util/automation.py [hunk 1]
```diff
@@ -115,14 +135,20 @@ def get_new_commits(sha1_from):
...
         # When there is no change, just nicely output no new commits
         return [], r.status_code
     elif payload['status'] == 'ahead':
         # Let us just keep the merges commits, we can identify them because they
         # are the commits with two parents
-        merges = filter(lambda x: len(x['parents']) >= 2, payload['commits'])
-        return map(lambda x: x['sha'], merges), r.status_code
+        all_merges = filter(lambda x: len(x['parents']) >= 2, payload['commits'])
+        # Keep only merges made from Github UI
+        gh_merges = filter(lambda x: is_webflow(x), all_merges)
```
Comment (tilmankamp) [inline util/automation.py]: Why just the web ones?

Comment (lissyx) [inline util/automation.py]: Because that is the simplest way I found to identify a merge from a PR rather than a merge that you could have done locally :)

Comment (tilmankamp) [inline util/automation.py]: I see. But what's so important on only testing/running PRs?

Comment (lissyx) [inline util/automation.py]: We don't want to waste resources, a PR that contains 5 commits would trigger 5 trainings, each running for at least 2-3h. Hence why trying to limit this to merges from PR.

Comment (tilmankamp) [inline util/automation.py]: Got it.

**Pull request merged**